packages/in_app_purchase/lib/src/store_kit_wrappers/sk_payment_queue_wrapper.dart

```diff
@@ -14,9 +16,174 @@ part 'sk_payment_queue_wrapper.g.dart';
+  /// started a purchase flow from the App Store.
+  void setTransactionObserver(SKTransactionObserverWrapper observer) {
+    _observer = observer;
+  }
+
+  /// Adds a payment object to the payment queue.
+  ///
+  /// Prior to this call, you have to make sure At least one [SKTransactionObserverWrapper] should be added to the payment queue
+  /// using [addTransactionObserver]. You also have to make sure the [SKProductWrapper] of the payment has been fetched using [SKRequestMaker.startProductRequest].
+  /// Each payment will generate a [SKPaymentTransactionWrapper]. After a payment is being added to the payment queue, the [SKTransactionObserverWrapper] is responsible to handle
+  /// the transaction that is generated by the payment.
+  /// The [productIdentifier] must match one of the product that is returned in [SKRequestMaker.startProductRequest].
+  /// to test the payment in the [sandbox](https://developer.apple.com/apple-pay/sandbox-testing/).
+  Future<void> addPayment(SKPaymentWrapper payment) async {
+    assert(_observer != null,
+        '[in_app_purchase]: Trying to add a payment without an observer. Observer must be set using `setTransactionObserver` before adding a payment `setTransactionObserver`. It is mandatory to set the observer at the moment when app launches.');
+    Map requestMap = payment.toMap();
+    await channel.invokeMethod(
+      '-[InAppPurchasePlugin addPayment:result:]',
+      requestMap,
+    );
+  }
+
+  /// Finishes a transaction, remove it from the queue.
+  ///
+  /// This method should be called from a observer callback when receiving notification from the payment queue. You should only
+  /// call this method after the transaction is successfully processed and the functionality purchased by the user is unlocked.
+  /// Itt will throw a Platform exception if the [SKPaymentTransactionWrapper.transactionState] is [SKPaymentTransactionStateWrapper.purchasing].
+  Future<void> finishTransaction(
+      SKPaymentTransactionWrapper transaction) async {
+    await channel.invokeMethod(
+        '-[InAppPurchasePlugin finishTransaction:result:]',
+        transaction.transactionIdentifier);
+  }
+
+  // Triage a method channel call from the platform and triggers the correct observer method.
+  Future<dynamic> _handleObserverCallbacks(MethodCall call) {
+    assert(_observer != null,
+        'in_app_purchase]: (Fatal)The observer has not been set but we received a purchase transaction notification. Please ensure the observer has been set using `setTransactionObserver`. Make sure the observer is added right at the App Launch.');
+    switch (call.method) {
+      case 'updatedTransactions':
+        {
+          final List<SKPaymentTransactionWrapper> transactions =
+              _getTransactionList(call.arguments);
+          return Future<void>(() {
+            _observer.updatedTransactions(transactions: transactions);
+          });
```

Comment:
I think we want `break`s after each case?

Reply:
there should be a return already for each case.